anonymous_helpline_chatbot/telegram_bot/tg_callback_handlers.py

```diff
@@ -47,23 +47,28 @@ def conversation_rate_callback_query(call: telebot.types.CallbackQuery):
 @nonfalling_handler
 def conversation_acceptation_callback_query(call: telebot.types.CallbackQuery):
     d = jload_and_expand_callback_data(call.data)
-    # TODO: There are a private member access (`core._operators_invitations_messages`) and a race condition
-    #  (conversation can begin after the if statement) here. Both will be fixed with #37
-    if call.message.chat.id in core._operators_invitations_messages.keys():
-        bot.answer_callback_query(call.id, "Невозможно начать беседу, пока вы ожидаете оператора")
-        return
 
-    conversation_began = core.begin_conversation(d['client_id'], call.message.chat.id)
-
-    if conversation_began:
-        core.clear_invitation_messages(d['client_id'])
-
-        (_, local_client_id), (_, local_operator_id) = core.get_conversing(call.message.chat.id)
-        bot.answer_callback_query(call.id)
-        bot.send_message(call.message.chat.id, f"Началась беседа с клиентом №{local_client_id}. Отправьте "
-                                               "сообщение, и собеседник его увидит")
-        bot.send_message(d['client_id'], f"Началась беседа с оператором №{local_operator_id}. Отправьте сообщение, "
-                                         "и собеседник его увидит")
-    else:
-        bot.answer_callback_query(call.id, "Что-то пошло не так. Возможно, вы уже в беседе, или другой оператор принял "
-                                           "это приглашение")
+    with core.begin_conversation_with_locking(d['client_id'], call.message.chat.id) as result:
+        if result == 0:
+            local_client_id = core.get_local_id(d['client_id'])
+            local_operator_id = core.get_local_id(call.message.chat.id)
+            bot.send_message(call.message.chat.id, f"Началась беседа с клиентом №{local_client_id}. Отправьте "
+                                                   "сообщение, и собеседник его увидит")
+            bot.send_message(d['client_id'], f"Началась беседа с оператором №{local_operator_id}. Отправьте сообщение, "
+                                             "и собеседник его увидит")
+            bot.answer_callback_query(call.id)
+        elif result == 1:
+            notify_admins(text="Consistency error: someone is trying to accept an invitation, where a client is "
+                               "operating!")
+            bot.send_message(call.message.chat.id, "У нас серьезные технические проблемы. Я уже уведомил разработчика")
```

Comment:
I don't think we need to react to these consistency errors that aggressively, because probably, for example, someone has quick fingers and managed to click on an invitation before it should have been removed? I mean, it's very unlikely, so admins should be notified, but there should probably need a note about this, and the users probably shouldn't receive _messages_ about this - just a callback query answer would be enough in this case